sqlparser/test/scanner/single_token_test.dart

```diff
@@ -74,14 +74,33 @@ void main() {
     expect(scanner.tokens[1].type, TokenType.eof);
   });
 
-  test('issues error for unterminated string literals', () {
-    final scanner = Scanner("'unterminated")..scanTokens();
+  group("parse unterminated string literals", () {
+    test('issues error', () {
+      final scanner = Scanner("'unterminated")..scanTokens();
+
+      expect(
+        scanner.errors,
+        contains(const TypeMatcher<TokenizerError>()
+            .having((e) => e.message, 'message', 'Unterminated string')),
+      );
 
-    expect(
-      scanner.errors,
-      contains(const TypeMatcher<TokenizerError>()
-          .having((e) => e.message, 'message', 'Unterminated string')),
-    );
+      expect(scanner.tokens, isNotEmpty);
+      expect(scanner.tokens[0], isA<StringLiteralToken>());
+      expect(
+        (scanner.tokens[0] as StringLiteralToken).value,
+        equals("unterminated"),
+      );
```

Comment:
Nit: Use a single `expect` call:

```suggestion
      expect(scanner.tokens, [
        isA<StringLiteralToken>()
            .having((e) => e.value, 'value', 'unterminated'),
        isA<Token>().having((e) => e.type, 'type', TokenType.eof),
      ]);
```

Reply:
I added it to the commit